acme/acme/messages.py

```diff
@@ -172,6 +172,7 @@ class Directory(jose.JSONDeSerializable):
     class Meta(jose.JSONObjectWithFields):
         """Directory Meta."""
         terms_of_service = jose.Field('terms-of-service', omitempty=True)
+        terms_of_service_v2 = jose.Field('termsOfService', omitempty=True)
```

Comment:
This can be something we delay until closer until the release if you want, but I wonder if we shouldn't try and make the `terms_of_service` field automatically handle the difference in naming conventions between v1 and v2 for the user of the class and remove `terms_of_service_v2`. We did this for challenges in #5300 which was more complicated since we also send challenges back to the server which isn't something we do for directories.

Reply:
since only the directory fields in `Meta` are explicitly defined, the one class works for both; I'd rather not have to have `Directory` also be aware of acme version if possible.

Reply:
I just think it's a chance to keep the API cleaner/simpler and push details further down so the higher levels don't have to worry about them. If you didn't already, take a look at the approach in #5300 at a high level (protected fields with a property that just returns whichever value was set by the server). I personally favor that approach, but if you don't we can leave this as is.